aten/src/THCUNN/generic/MultiLabelMarginCriterion.cu

```diff
@@ -58,49 +58,52 @@ void THNN_(MultiLabelMarginCriterion_updateOutput)(
-        <<<blocks, threads, 0, c10::cuda::getCurrentCUDAStream()>>>(
-          THCTensor_(data)(state, output_tmp),
-          THCTensor_(data)(state, input),
-          THCIndexTensor_(data)(state, target),
-          THCTensor_(data)(state, istarget),
-          nframe, dim,
-          reduction == at::Reduction::Mean
-          );
-      THCudaCheck(cudaGetLastError());
-      auto t = THTensor_wrap(output_tmp);
-      auto r = THTensor_wrap(output);
-      at::native::sum_out(r, t, at::IntArrayRef(std::vector<int64_t>{}), false, r.scalar_type());
-      THCTensor_(free)(state, output_tmp);
-    }
-    else
-    {
-      THCTensor_(resize1d)(state, output, input->size(0));
-
-      cunn_MultiLabelMarginCriterion_updateOutput_kernel<scalar_t, accreal>
-        <<<blocks, threads, 0, c10::cuda::getCurrentCUDAStream()>>>(
-          THCTensor_(data)(state, output),
-          THCTensor_(data)(state, input),
-          THCIndexTensor_(data)(state, target),
-          THCTensor_(data)(state, istarget),
-          nframe, dim,
-          false
-          );
-      THCudaCheck(cudaGetLastError());
+    if (input->numel() != 0) {
+      int nframe = input->size(0);
+      int dim = input->size(1);
+      dim3 blocks(input->size(0));
+      dim3 threads(MULTILABELMARGIN_THREADS);
+
+      if (reduction != at::Reduction::None)
+      {
+        THCTensor *output_tmp = THCTensor_(newWithSize1d)(state, input->size(0));
+        THCTensor_(resize0d)(state, output);
+
+        cunn_MultiLabelMarginCriterion_updateOutput_kernel<scalar_t, accreal>
+          <<<blocks, threads, 0, c10::cuda::getCurrentCUDAStream()>>>(
+            THCTensor_(data)(state, output_tmp),
+            THCTensor_(data)(state, input),
+            THCIndexTensor_(data)(state, target),
+            THCTensor_(data)(state, istarget),
+            nframe, dim,
+            reduction == at::Reduction::Mean
+            );
+        THCudaCheck(cudaGetLastError());
+        auto t = THTensor_wrap(output_tmp);
+        auto r = THTensor_wrap(output);
+        at::native::sum_out(r, t, at::IntArrayRef(std::vector<int64_t>{}), false, r.scalar_type());
+        THCTensor_(free)(state, output_tmp);
+      }
+      else
+      {
+        THCTensor_(resize1d)(state, output, input->size(0));
+
+        cunn_MultiLabelMarginCriterion_updateOutput_kernel<scalar_t, accreal>
+          <<<blocks, threads, 0, c10::cuda::getCurrentCUDAStream()>>>(
+            THCTensor_(data)(state, output),
+            THCTensor_(data)(state, input),
+            THCIndexTensor_(data)(state, target),
+            THCTensor_(data)(state, istarget),
+            nframe, dim,
+            false
+            );
+        THCudaCheck(cudaGetLastError());
+      }
     }
   }
   else {
-    TORCH_INTERNAL_ASSERT(false, "non-empty vector or matrix expected (shouldn't get here)");
+    TORCH_INTERNAL_ASSERT(false, "Expected 2D input with non-zero batch dim, or 1D input with non-zero dim, but got sizes: ", 
```

Comment:
This doesn't seem correct because 2D can have zero batch dim (that's this PR) and 0D tensors also seem OK?

Maybe:

"Expected a tensor with two or fewer dimensions, but got X!"

Also TORCH_CHECK instead of TORCH_INTERNAL_ASSERT.